docs/cyber.md

```diff
@@ -0,0 +1,80 @@
+- [Talk at European Spark Conference 2019](https://databricks.com/session_eu19/cybermltoolkit-anomaly-detection-as-a-scalable-generic-service-over-apache-spark)
+- [(Internal Microsoft) Talk at MLADS November 2018](https://resnet.microsoft.com/video/42395)
+- [(Internal Microsoft) Talk at MLADS June 2019](https://resnet.microsoft.com/video/43618)
+
+1. [ComplementAccessTransformer](../src/main/python/mmlspark/cyber/ml/access_anomalies/complement_access.py)
+   is a SparkML [Transformer](https://spark.apache.org/docs/2.2.0/api/java/index.html?org/apache/spark/ml/Transformer.html).
+   Given a dataframe it returns a new dataframe containing new access patterns sampled from
+   the set of possible access patterns which did not occur in the given dataframe
+   (i.e., it returns a sample from the complement set).
+
+# feature engineering: [indexers.py](../src/main/python/mmlspark/cyber/ml/feature_engineering/indexers.py)
+1. [IdIndexer](../src/main/python/mmlspark/cyber/ml/feature_engineering/indexers.py)
+   is a SparkML [Estimator](https://spark.apache.org/docs/2.2.0/api/java/index.html?org/apache/spark/ml/Estimator.html).
+   Given a dataframe, it creates an IdIndexerModel (described next) for categorical features which
+   contains the information to map each partition and column seen in the given dataframe to an id.
+   for each partition or one consecutive range for all partition and column values.
+2. [IdIndexerModel](../src/main/python/mmlspark/cyber/ml/feature_engineering/indexers.py)
+   is a SparkML [Transformer](https://spark.apache.org/docs/2.2.0/api/java/index.html?org/apache/spark/ml/Transformer.html).
+   Given a dataframe maps each partition and column field to a consecutive integer id.
+   Partitions or column values not encountered in the estimator are mapped to 0.
+   The model can operate in two modes, either create consecutive integer id independently 
+3. [MultiIndexer](../src/main/python/mmlspark/cyber/ml/feature_engineering/indexers.py)
+   is a SparkML [Estimator](https://spark.apache.org/docs/2.2.0/api/java/index.html?org/apache/spark/ml/Estimator.html).
+   Uses multiple IdIndexer to generate a MultiIndexerModel (described next) for categorical features which
+   contains multiple IdIndexers for multiple partitions and columns.
+4. [MultiIndexerModel](../src/main/python/mmlspark/cyber/ml/feature_engineering/indexers.py)
+   is a SparkML [Transformer](https://spark.apache.org/docs/2.2.0/api/java/index.html?org/apache/spark/ml/Transformer.html).
+   Given a dataframe maps each partition and column field to a consecutive integer id.
+   Partitions or column values not encountered in the estimator are mapped to 0.
+   The model can operate in two modes, either create consecutive integer id independently 
+
+# feature engineering: [scalers.py](../src/main/python/mmlspark/cyber/ml/feature_engineering/scalers.py)
+1. [StandardScalarScaler](../src/main/python/mmlspark/cyber/ml/feature_engineering/scalers.py)
```

Comment:
How is this different than https://spark.apache.org/docs/latest/mllib-feature-extraction.html#standardscaler?

Reply:
The spark scalars operate on spark vector objects whereas this component works on scalars, hence the name StandardScalarScaler ... In addition, my scalar supports a scenario where the data belongs to multiple tenants and each tenant should be scaled independently.